server/src/test/java/org/opensearch/index/remote/RemoteStoreStatsTrackerFactoryTests.java

```diff
@@ -0,0 +1,128 @@
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        shardId = new ShardId("index", "uuid", 0);
+        indexShard = createIndexShard(shardId, true);
+        remoteStoreStatsTrackerFactory = new RemoteStoreStatsTrackerFactory(Settings.EMPTY);
+        shardId2 = null;
+        indexShard2 = null;
+    }
+
+    public void testAfterIndexShardCreatedForRemoteBackedIndex() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNotNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(indexShard.shardId()));
+    }
+
+    public void testAfterIndexShardCreatedForNonRemoteBackedIndex() {
+        indexShard = createIndexShard(shardId, false);
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(indexShard.shardId()));
+    }
+
+    public void testAfterIndexShardClosed() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        assertNotNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId));
+        remoteStoreStatsTrackerFactory.afterIndexShardClosed(shardId, indexShard, indexShard.indexSettings().getSettings());
+        assertNull(remoteStoreStatsTrackerFactory.getRemoteSegmentTransferTracker(shardId));
+    }
+
+    public void testUpdateMovingAverageWindowSizeZero() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        createSecondShard();
+
+        int defaultSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE;
+        assertDefaultSize(defaultSize);
+
+        int updatedSize = 0;
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+                RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+                updatedSize
+            )
+        );
+        assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
+    }
+
+    public void testUpdateMovingAverageWindowSizeLessThanMinAllowed() {
+        remoteStoreStatsTrackerFactory.afterIndexShardCreated(indexShard);
+        createSecondShard();
+
+        int defaultSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE;
+        assertDefaultSize(defaultSize);
+
+        int updatedSize = RemoteStorePressureSettings.Defaults.MOVING_AVERAGE_WINDOW_SIZE_MIN_VALUE - 1;
+        // TODO: The following exception isn't thrown currently
+        // assertThrows(
+        // IllegalArgumentException.class,
+        // () -> remoteStoreStatsTrackerFactory.updateMovingAverageWindowSize(
+        // RemoteSegmentTransferTracker::updateMovingAverageWindowSize,
+        // updatedSize
+        // )
+        // );
+        // assertEquals(defaultSize, remoteStoreStatsTrackerFactory.getMovingAverageWindowSize());
```

Comment:
The minimum validation exists on the setting. When supplying the value here, the validation has already been done at the cluster service layer. Please refer https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/index/remote/RemoteStorePressureSettingsTests.java for writing tests for min validation.